<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      Enrichment between genomic regions &middot; Hippocamplus
    
  </title>

  
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-113785126-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    
    gtag('config', "UA-113785126-1");
  </script>
  
  
  <link rel="stylesheet" href="../../../../css/poole.css">
  <link rel="stylesheet" href="../../../../css/syntax.css">
  <link rel="stylesheet" href="../../../../css/lanyon.css">
  <link rel="stylesheet" href="../../../../css/idea.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=PT+Serif:400,400italic,700|PT+Sans:400">
  <link rel="stylesheet" href="../../../../fa/css/font-awesome.min.css">
  
  
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="../../../../assets/hippoPlus-icon.png">
  <link rel="shortcut icon" href="../../../../assets/hippoPlus-icon.png">

  
  <link rel="alternate" type="application/rss+xml" title="RSS" href="../../../../atom.xml">
</head>

  
  <body>

    
<input type="checkbox" class="sidebar-checkbox" id="sidebar-checkbox">


<div class="sidebar" id="sidebar">
  <div class="sidebar-item">
    <p>A place to gather useful information I keep on forgetting.</p>
  </div>

  <nav class="sidebar-nav">
    <a class="sidebar-nav-item " href="../../../../">Home</a>
    <a class="sidebar-nav-item " href="../../../../post">Posts</a>

    
    
      
        <a class="sidebar-nav-item " href="../../../../emacs/">Emacs</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../expression/">Expression</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../latex/">LaTeX</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../markdown/">Markdown</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../math/stats/">Math/Stats</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../org-mode/">Org mode</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../python/">Python</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../r/">R</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../shell/">Shell</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../internet/">Internet</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../about/">About</a>
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
        <a class="sidebar-nav-item " href="../../../../ryan-lorig-roach/">Ryan Lorig-Roach</a>
      
    

    
    
  </nav>

  <div class="sidebar-item">
  </div>
  
  <div class="sidebar-item">
    <div><a href="../../../../index.xml"><i class="fa fa-rss-square" aria-hidden="true"></i> RSS</a></div>
    <div><a href="https://twitter.com/JMonlong"><i class="fa fa-twitter" aria-hidden="true"></i> @JMonlong</a></div>
    <p>&copy; 2019 Jean Monlong. All rights reserved.</p>
  </div>
</div>


    
    <div class="wrap">
      <div class="masthead">
        <div class="container">
          <h3 class="masthead-title">
            <a href="../../../../" title="Home">Hippocamplus</a>
            <small>My Second Memory</small>
          </h3>
        </div>
      </div>

      <div class="container content">


<div class="post">
  <h1 class="post-title">Enrichment between genomic regions</h1>
  <span class="post-date">Sep 5 2017
  
  
  
  
  
  <a href='../../../../tags/genome/' class='label'>genome</a>
  
  <a href='../../../../tags/stats/' class='label'>stats</a>
  
  
  
  </span>
  


<p>Testing if two sets of genomic regions overlap significantly is not straightforward. In the simple situation of regions of 1 bp (e.g. SNVs) we could use a hypergeometric test. When the regions are small enough and there are not too many, the hypergeometric test might also be a fair approximation.</p>
<p>But when we manipulate many regions of variable size covering the entire genome it’s not as straightforward. The gene annotation is an example. The repeat annotation is even worse as it covers almost 50% of the genome and contains different families with very different size/location profiles.</p>
<pre class="r"><code>library(ggplot2)
library(dplyr)
library(magrittr)
library(broom)
library(knitr)
library(tidyr)</code></pre>
<div id="simulated-data" class="section level2">
<h2>Simulated data</h2>
<p>In a very simple scenario of having only one chromosome of size 250 Mbp.</p>
<p>First let’s create a function that draw random regions (ranges) in this chromosome.</p>
<pre class="r"><code>library(IRanges)
randRegions &lt;- function(sizes, max.pos = 2.5e+08, max.iter = 10) {
    gr = IRanges(runif(length(sizes), 0, max.pos - 
        sizes), width = sizes)
    dup = which(countOverlaps(gr, gr) &gt; 1)
    iter = 1
    while (iter &lt;= max.iter &amp; length(dup) &gt; 0) {
        gr[dup] = IRanges(runif(length(dup), 0, max.pos - 
            sizes[dup]), width = sizes[dup])
        dup = which(countOverlaps(gr, gr) &gt; 1)
    }
    return(gr)
}</code></pre>
<p>Now some regions will be our “repeats”: 10,000 regions from size 10 bp to 6 Kbp.</p>
<pre class="r"><code>rep.r = randRegions(runif(10000, 10, 6000))
sum(width(rep.r))/2.5e+08</code></pre>
<pre><code>## [1] 0.1196666</code></pre>
<p>They cover 11.97% of the chromosome.</p>
<p>Now if we have another set of regions and we want to know how much they overlap with the repeats we could use the hypergeometric test. With this test we assume that we are sampling bases in the genome and testing if it’s covered by a repeat. In that sense, we expect 11.97% of our regions to overlap a repeat. If we compare random regions there shouldn’t be a significant overlap and the distribution of the P-value should be flat.</p>
<pre class="r"><code>testHG &lt;- function(feat.r, nb = 1000, size = 1, nb.test = 3000, 
    total.b = 2.5e+08) {
    exp.b = sum(width(feat.r))
    sapply(1:nb.test, function(ii) {
        reg.r = randRegions(rep(size, nb))
        obs.ol = sum(overlapsAny(reg.r, feat.r))
        phyper(obs.ol, exp.b, total.b - exp.b, length(reg.r), 
            lower.tail = FALSE)
    })
}

ht.sim = rbind(data.frame(nb = 1000, size = 1, pv = testHG(rep.r, 
    1000, 1)), data.frame(nb = 1000, size = 1000, pv = testHG(rep.r, 
    1000, 1000)), data.frame(nb = 100, size = 1000, 
    pv = testHG(rep.r, 100, 1000)), data.frame(nb = 1000, 
    size = 100, pv = testHG(rep.r, 1000, 100)))</code></pre>
<pre class="r"><code>ht.sim %&gt;% mutate(nbsize = paste0(nb, &quot; x &quot;, size, 
    &quot;bp&quot;)) %&gt;% group_by(nbsize) %&gt;% arrange(pv) %&gt;% 
    mutate(cumprop = (1:n())/n()) %&gt;% ggplot(aes(x = pv, 
    y = cumprop, color = nbsize)) + geom_line() + theme_bw() + 
    geom_abline(linetype = 2) + ylab(&quot;cumulative proportion&quot;) + 
    xlab(&quot;P-value&quot;) + scale_color_brewer(palette = &quot;Set1&quot;, 
    name = &quot;regions&quot;) + theme(legend.justification = c(1, 
    0), legend.position = c(0.99, 0.01))</code></pre>
<p><img src="../../../../post/2017-09-05-GenomicRegionEnrichment_files/figure-html/testhggraph-1.png" width="672" /></p>
<p>As expected, the hypergeometric test works well for region of 1 bp. Otherwise the distribution of the P-values is biased. The larger the regions the stronger the bias. To a lower extent, more regions also means more bias.</p>
</div>
<div id="using-control-regions-with-similar-features" class="section level2">
<h2>Using control regions with similar features</h2>
<p>We want to control for the size distribution and the total number of regions tested. Instead of the hypergeometric test, we can get control regions and compare their overlap with the actual regions, using a logistic regression for example. The control regions must be randomly distributed in the genome but have the same size distribution as our original regions. In the logistic regression we compare the two binary variables: overlapping a repeat or not, being an original region or a control region.</p>
<pre class="r"><code>testLR &lt;- function(feat.r, nb = 1000, size = 1, nb.test = 3000) {
    sapply(1:nb.test, function(ii) {
        reg.r = randRegions(rep(size, nb))
        cont.r = randRegions(width(reg.r))
        df = rbind(data.frame(region = TRUE, ol = overlapsAny(reg.r, 
            feat.r)), data.frame(region = FALSE, ol = overlapsAny(cont.r, 
            feat.r)))
        pvs = glm(ol ~ region, data = df, family = binomial()) %&gt;% 
            tidy %&gt;% .$p.value
        pvs[2]
    })
}

lr.sim = rbind(data.frame(nb = 1000, size = 1, pv = testLR(rep.r, 
    1000, 1)), data.frame(nb = 1000, size = 1000, pv = testLR(rep.r, 
    1000, 1000)), data.frame(nb = 100, size = 1000, 
    pv = testLR(rep.r, 100, 1000)), data.frame(nb = 1000, 
    size = 100, pv = testLR(rep.r, 1000, 100)))</code></pre>
<pre class="r"><code>lr.sim %&gt;% mutate(nbsize = paste0(nb, &quot; x &quot;, size, 
    &quot;bp&quot;)) %&gt;% group_by(nbsize) %&gt;% arrange(pv) %&gt;% 
    mutate(cumprop = (1:n())/n()) %&gt;% ggplot(aes(x = pv, 
    y = cumprop, color = nbsize)) + geom_line() + theme_bw() + 
    geom_abline(linetype = 2) + ylab(&quot;cumulative proportion&quot;) + 
    xlab(&quot;P-value&quot;) + scale_color_brewer(palette = &quot;Set1&quot;, 
    name = &quot;regions&quot;) + theme(legend.justification = c(1, 
    0), legend.position = c(0.99, 0.01))</code></pre>
<p><img src="../../../../post/2017-09-05-GenomicRegionEnrichment_files/figure-html/contreggraph-1.png" width="672" /></p>
<p>The distribution of the P-values is much better.</p>
</div>
<div id="controlling-for-correlated-features" class="section level2">
<h2>Controlling for correlated features</h2>
<p>In the genome, the distribution of genes, repeats, functional regions, and others is not random. Different types of elements tend to be found together while others don’t. For example some repeats are located in GC-rich regions and others in AT-rich regions. Transposable elements don’t overlap exonic regions much. Their are hotspots of segmental duplications.</p>
<p>Sometimes we want to control for the overlap with one (or more) genomic features to test the independent association of another. For example, we known copy number variants (CNVs) are enriched in segmental duplications and transposable elements are also enriched in segmental duplications. We might want to test if CNVs are independently enriched in regions with transposable elements, controlling for the overlap with segmental duplications.</p>
<p>I tried to simulate a first set of regions that significantly overlaps our repeats and another one that significantly overlaps the first set. That way we should see a significant overlap with repeat when we test them separately, but the second one should be significant when we control for the first one.</p>
<pre class="r"><code>corRegions &lt;- function(sizes, feat.r, or = 2, max.iter = 10, 
    max.pos = 2.5e+08) {
    reg.r = randRegions(sizes)
    for (ii in 1:or) {
        reg.r = c(reg.r[overlapsAny(reg.r, feat.r)], 
            randRegions(sizes))
    }
    dup = which(countOverlaps(reg.r, reg.r) &gt; 1)
    sizes = width(reg.r)
    iter = 1
    while (iter &lt;= max.iter &amp; length(dup) &gt; 0) {
        reg.r[dup] = IRanges(runif(length(dup), 0, 
            max.pos - sizes[dup]), width = sizes[dup])
        dup = which(countOverlaps(reg.r, reg.r) &gt; 1)
    }
    reg.r
}

## First set of regions
repcor.r = corRegions(rep(10000, 1000), rep.r)
cont.r = randRegions(width(repcor.r))
df = rbind(data.frame(region = TRUE, ol = overlapsAny(repcor.r, 
    rep.r)), data.frame(region = FALSE, ol = overlapsAny(cont.r, 
    rep.r)))
glm(ol ~ region, data = df, family = binomial()) %&gt;% 
    tidy %&gt;% kable</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">term</th>
<th align="right">estimate</th>
<th align="right">std.error</th>
<th align="right">statistic</th>
<th align="right">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">(Intercept)</td>
<td align="right">-0.2375323</td>
<td align="right">0.0464770</td>
<td align="right">-5.11075</td>
<td align="right">3e-07</td>
</tr>
<tr class="even">
<td align="left">regionTRUE</td>
<td align="right">0.8549335</td>
<td align="right">0.0670786</td>
<td align="right">12.74525</td>
<td align="right">0e+00</td>
</tr>
</tbody>
</table>
<pre class="r"><code>## Second set of regions
seed.r = randRegions(rep(10000, 6000))
seed.ol = overlapsAny(seed.r, repcor.r)
repcorcor.r = c(seed.r[seed.ol], seed.r[head(which(!seed.ol), 
    sum(!seed.ol) * 0.05)])
cont.r = randRegions(width(repcorcor.r))
df = rbind(data.frame(region = TRUE, ol = overlapsAny(repcorcor.r, 
    rep.r)), data.frame(region = FALSE, ol = overlapsAny(cont.r, 
    rep.r)))
glm(ol ~ region, data = df, family = binomial()) %&gt;% 
    tidy %&gt;% kable</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">term</th>
<th align="right">estimate</th>
<th align="right">std.error</th>
<th align="right">statistic</th>
<th align="right">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">(Intercept)</td>
<td align="right">-0.2231436</td>
<td align="right">0.0590620</td>
<td align="right">-3.778125</td>
<td align="right">0.0001580</td>
</tr>
<tr class="even">
<td align="left">regionTRUE</td>
<td align="right">0.2214209</td>
<td align="right">0.0832684</td>
<td align="right">2.659123</td>
<td align="right">0.0078344</td>
</tr>
</tbody>
</table>
<div id="extending-the-logistic-regression-model" class="section level3">
<h3>Extending the logistic regression model</h3>
<p>One strategy is to add a variable in the model that represents the effect we want to control for.</p>
<pre class="r"><code>df = rbind(data.frame(region = TRUE, ol = overlapsAny(repcorcor.r, 
    rep.r), repcor = overlapsAny(repcorcor.r, repcor.r)), 
    data.frame(region = FALSE, ol = overlapsAny(cont.r, 
        rep.r), repcor = overlapsAny(cont.r, repcor.r)))
glm(ol ~ region + repcor, data = df, family = binomial()) %&gt;% 
    tidy %&gt;% kable</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">term</th>
<th align="right">estimate</th>
<th align="right">std.error</th>
<th align="right">statistic</th>
<th align="right">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">(Intercept)</td>
<td align="right">-0.2817899</td>
<td align="right">0.0612109</td>
<td align="right">-4.6035936</td>
<td align="right">0.0000042</td>
</tr>
<tr class="even">
<td align="left">regionTRUE</td>
<td align="right">-0.0452678</td>
<td align="right">0.1093757</td>
<td align="right">-0.4138742</td>
<td align="right">0.6789663</td>
</tr>
<tr class="odd">
<td align="left">repcorTRUE</td>
<td align="right">0.4156934</td>
<td align="right">0.1096425</td>
<td align="right">3.7913531</td>
<td align="right">0.0001498</td>
</tr>
</tbody>
</table>
<p>As expected, adding a variable in the model controls for this.</p>
</div>
<div id="better-control-regions" class="section level3">
<h3>Better control regions</h3>
<p>Another approach is to control the specific overlap in the control regions. We want to force our control regions to overlap as much with the feature as the original regions.</p>
<pre class="r"><code>randRegionsCons &lt;- function(reg.r, feat.r, nb.seed = 1e+06) {
    seed.r = randRegions(rep(1, nb.seed))
    dist.df = distanceToNearest(seed.r, feat.r) %&gt;% 
        as.data.frame
    reg.ol = overlapsAny(reg.r, feat.r)
    res.r = lapply(unique(width(reg.r)), function(size) {
        size.ii = which(width(reg.r) == size)
        res.r = IRanges()
        if (sum(reg.ol[size.ii]) &gt; 0) {
            seed.ii = dist.df %&gt;% filter(distance &lt; 
                size/2) %&gt;% .$queryHits %&gt;% sample(sum(reg.ol[size.ii]))
            res.r = c(res.r, resize(seed.r[seed.ii], 
                size, fix = &quot;center&quot;))
        }
        if (sum(!reg.ol[size.ii]) &gt; 0) {
            seed.ii = dist.df %&gt;% filter(distance &gt; 
                size/2) %&gt;% .$queryHits %&gt;% sample(sum(!reg.ol[size.ii]))
            res.r = c(res.r, resize(seed.r[seed.ii], 
                size, fix = &quot;center&quot;))
        }
        res.r
    })
    do.call(c, res.r)
}

contSize.r = randRegionsCons(repcorcor.r, repcor.r)
df = rbind(data.frame(region = TRUE, ol = overlapsAny(repcorcor.r, 
    rep.r), repcor = overlapsAny(repcorcor.r, repcor.r)), 
    data.frame(region = FALSE, ol = overlapsAny(contSize.r, 
        rep.r), repcor = overlapsAny(contSize.r, repcor.r)))
glm(ol ~ region, data = df, family = binomial()) %&gt;% 
    tidy %&gt;% kable</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">term</th>
<th align="right">estimate</th>
<th align="right">std.error</th>
<th align="right">statistic</th>
<th align="right">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">(Intercept)</td>
<td align="right">0.1709578</td>
<td align="right">0.0589111</td>
<td align="right">2.901961</td>
<td align="right">0.0037083</td>
</tr>
<tr class="even">
<td align="left">regionTRUE</td>
<td align="right">-0.1726805</td>
<td align="right">0.0831615</td>
<td align="right">-2.076448</td>
<td align="right">0.0378526</td>
</tr>
</tbody>
</table>
<p>It works too. One benefit of this approach is its interpretability: we can directly compare summary metrics using the control regions, e.g. like the proportion of regions overlapping repeats.</p>
<pre class="r"><code>mean(overlapsAny(repcorcor.r, repcor.r))</code></pre>
<pre><code>## [1] 0.7812231</code></pre>
<pre class="r"><code>mean(overlapsAny(cont.r, repcor.r))</code></pre>
<pre><code>## [1] 0.1395349</code></pre>
<pre class="r"><code>mean(overlapsAny(contSize.r, repcor.r))</code></pre>
<pre><code>## [1] 0.7812231</code></pre>
<p>This could be useful in situation with extreme overlap distribution. If only a few regions overlap the feature in the simple control regions, the regression might not correct for it as well as if forcing the control regions to be similar. Maybe there would some differences in power in those cases ?</p>
<p>However building these control regions can become computationally intense, especially if the sizes of the regions vary and several features need to be controlled.</p>
<p>In practice I would do both: include the variable in the regression model and use regions with controlled overlap.</p>
</div>
</div>
<div id="comparing-different-sets" class="section level2">
<h2>Comparing different sets</h2>
<p>What if we need to compare sets of regions <em>A</em> and <em>B</em> with a third one <em>C</em>. If the <em>A</em> and <em>B</em> are comparable in term of size and total number we could directly compare the overlap or an enrichment estimate (e.g. model estimate). If <em>A</em> and <em>B</em> have different size distribution or just total number of regions, these estimates may not be directly comparable. If they both overlap significantly with <em>C</em>, the previous test (control regions + logistic regression) should test them significant. Even the P-value might be affected by the difference in size/number between the two sets. But how should we compared them ? Which interpretable metric could we use to compare enrichment of two different sets or regions ?</p>
<p>A practical example would be two catalogs of CNVs, say from two different methods, that we want to compare to a functional annotation. If one catalogs has more CNVs, or has larger CNVs, how can we say which one overlaps better with the functional annotation ?</p>
<p>I simulate this scenario and compare a few metrics:</p>
<ul>
<li>The fold-change in overlap proportion: proportion overlapping / proportion overlapping in control regions.</li>
<li>The diff-change in overlap proportion: proportion overlapping - proportion overlapping in control regions.</li>
<li>The logistic regression estimate which are log odds ratio.</li>
</ul>
<p>Well, the main question was how should I simulate this. I ended up simulating two sets with similar odds ratio so we already know which metric will work better… One of the value of simulation is to force us to define the question. Or at least think about it. In this example, forcing two different sets to have similar odds ratio seemed more natural than trying to double the proportion for example. The odds ratio seems more fair to me and might avoid the situation where we are more likely to observe a large effect size just because the regions are rarer/smaller.</p>
<p>Using a set of functional regions, I will try to compare a set of small CNVs and large CNVs. We expect more of the large CNVs to overlap the functional regions by chance.</p>
<pre class="r"><code>fun.r = randRegions(rep(10, 30000))
cnv.sm = randRegions(rep(1000, 1000))
cnv.lg = randRegions(rep(10000, 1000))
mean(overlapsAny(cnv.sm, fun.r))</code></pre>
<pre><code>## [1] 0.112</code></pre>
<pre class="r"><code>mean(overlapsAny(cnv.lg, fun.r))</code></pre>
<pre><code>## [1] 0.688</code></pre>
<pre class="r"><code>testLR &lt;- function(reg.r, feat.r) {
    cont.r = randRegions(width(reg.r))
    df.sm = rbind(data.frame(region = TRUE, ol = overlapsAny(reg.r, 
        feat.r)), data.frame(region = FALSE, ol = overlapsAny(cont.r, 
        feat.r)))
    rbind(data.frame(term = &quot;fold-change&quot;, estimate = mean(overlapsAny(reg.r, 
        feat.r))/mean(overlapsAny(cont.r, feat.r)), 
        p.value = NA), data.frame(term = &quot;diff-change&quot;, 
        estimate = mean(overlapsAny(reg.r, feat.r)) - 
            mean(overlapsAny(cont.r, feat.r)), p.value = NA), 
        glm(ol ~ region, data = df.sm, family = binomial()) %&gt;% 
            tidy %&gt;% select(term, estimate, p.value))
}

metrics.df = rbind(testLR(cnv.sm, fun.r) %&gt;% mutate(region = &quot;cnv.sm&quot;), 
    testLR(cnv.lg, fun.r) %&gt;% mutate(region = &quot;cnv.lg&quot;))
metrics.df %&gt;% filter(term != &quot;(Intercept)&quot;) %&gt;% select(region, 
    term, estimate) %&gt;% kable</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">region</th>
<th align="left">term</th>
<th align="right">estimate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">cnv.sm</td>
<td align="left">fold-change</td>
<td align="right">1.0566038</td>
</tr>
<tr class="even">
<td align="left">cnv.sm</td>
<td align="left">diff-change</td>
<td align="right">0.0060000</td>
</tr>
<tr class="odd">
<td align="left">cnv.sm</td>
<td align="left">regionTRUE</td>
<td align="right">0.0617938</td>
</tr>
<tr class="even">
<td align="left">cnv.lg</td>
<td align="left">fold-change</td>
<td align="right">0.9745042</td>
</tr>
<tr class="odd">
<td align="left">cnv.lg</td>
<td align="left">diff-change</td>
<td align="right">-0.0180000</td>
</tr>
<tr class="even">
<td align="left">cnv.lg</td>
<td align="left">regionTRUE</td>
<td align="right">-0.0852498</td>
</tr>
</tbody>
</table>
<div id="no-association" class="section level3">
<h3>No association</h3>
<p>If the CNVs are not enriched in the functional regions, how do the three metrics compare ?</p>
<pre class="r"><code>null.df = lapply(1:1000, function(ii) {
    cnv.sm = randRegions(rep(1000, 1000))
    cnv.lg = randRegions(rep(10000, 1000))
    rbind(testLR(cnv.sm, fun.r) %&gt;% mutate(region = &quot;cnv.sm&quot;, 
        rep = ii), testLR(cnv.lg, fun.r) %&gt;% mutate(region = &quot;cnv.lg&quot;, 
        rep = ii))
})
null.df = do.call(rbind, null.df)</code></pre>
<pre class="r"><code>null.df %&gt;% filter(term == &quot;regionTRUE&quot;) %&gt;% ggplot(aes(x = estimate, 
    colour = region)) + geom_density() + theme_bw() + 
    geom_vline(xintercept = 0) + ggtitle(&quot;log odds ratio&quot;)</code></pre>
<p><img src="../../../../post/2017-09-05-GenomicRegionEnrichment_files/figure-html/cnvnoassocgraph-1.png" width="672" /></p>
<pre class="r"><code>null.df %&gt;% filter(term == &quot;fold-change&quot;) %&gt;% ggplot(aes(x = estimate, 
    colour = region)) + geom_density() + theme_bw() + 
    geom_vline(xintercept = 1) + ggtitle(&quot;fold-change&quot;)</code></pre>
<p><img src="../../../../post/2017-09-05-GenomicRegionEnrichment_files/figure-html/cnvnoassocgraph-2.png" width="672" /></p>
<pre class="r"><code>null.df %&gt;% filter(term == &quot;diff-change&quot;) %&gt;% ggplot(aes(x = estimate, 
    colour = region)) + geom_density() + theme_bw() + 
    geom_vline(xintercept = 0) + ggtitle(&quot;diff-change&quot;)</code></pre>
<p><img src="../../../../post/2017-09-05-GenomicRegionEnrichment_files/figure-html/cnvnoassocgraph-3.png" width="672" /></p>
<p>The three metrics are centered in 0 but the variance of the fold-change metric is much higher for the small CNVs.</p>
</div>
<div id="association" class="section level3">
<h3>Association</h3>
<p>If the odds of overlapping the functional regions are doubled.</p>
<pre class="r"><code>asso.df = lapply(1:1000, function(ii) {
    cnv.sm = randRegions(rep(1000, 1000))
    cnv.lg = randRegions(rep(10000, 1000))
    cnv.sm = c(cnv.sm[overlapsAny(cnv.sm, fun.r)], 
        randRegions(rep(1000, 1000)))
    cnv.lg = c(cnv.lg[overlapsAny(cnv.lg, fun.r)], 
        randRegions(rep(10000, 1000)))
    rbind(testLR(cnv.sm, fun.r) %&gt;% mutate(region = &quot;cnv.sm&quot;, 
        rep = ii), testLR(cnv.lg, fun.r) %&gt;% mutate(region = &quot;cnv.lg&quot;, 
        rep = ii))
})
asso.df = do.call(rbind, asso.df)</code></pre>
<pre class="r"><code>asso.df %&gt;% filter(term == &quot;regionTRUE&quot;) %&gt;% ggplot(aes(x = estimate, 
    colour = region)) + geom_density() + theme_bw() + 
    geom_vline(xintercept = 0) + ggtitle(&quot;log odds ratio&quot;)</code></pre>
<p><img src="../../../../post/2017-09-05-GenomicRegionEnrichment_files/figure-html/cnvassocgraph-1.png" width="672" /></p>
<pre class="r"><code>asso.df %&gt;% filter(term == &quot;fold-change&quot;) %&gt;% ggplot(aes(x = estimate, 
    colour = region)) + geom_density() + theme_bw() + 
    geom_vline(xintercept = 1) + ggtitle(&quot;fold-change&quot;)</code></pre>
<p><img src="../../../../post/2017-09-05-GenomicRegionEnrichment_files/figure-html/cnvassocgraph-2.png" width="672" /></p>
<pre class="r"><code>asso.df %&gt;% filter(term == &quot;diff-change&quot;) %&gt;% ggplot(aes(x = estimate, 
    colour = region)) + geom_density() + theme_bw() + 
    geom_vline(xintercept = 0) + ggtitle(&quot;diff-change&quot;)</code></pre>
<p><img src="../../../../post/2017-09-05-GenomicRegionEnrichment_files/figure-html/cnvassocgraph-3.png" width="672" /></p>
<p>As expected by construction, only the logistic regression estimate are similar. If we used the fold-change metric it would look like the small CNVs are more enriched; with the diff-change metric the large CNVs would.</p>
</div>
</div>

</div>

      </div>
    </div>

    
    

    <label for="sidebar-checkbox" class="sidebar-toggle"></label>

    <script src="../../../../assets/highlight.pack.js"></script>
    <script src="../../../../assets/r.min.js"></script>
    
    <script>
      hljs.configure({languages: ['r']});
      hljs.initHighlightingOnLoad();
    </script>

    <script src="//yihui.name/js/math-code.js"></script>
    <script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
    
  </body>
</html>

