<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      First look at the gnomAD SV catalog &middot; Hippocamplus
    
  </title>

  
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-113785126-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    
    gtag('config', "UA-113785126-1");
  </script>
  
  
  <link rel="stylesheet" href="../../../../css/poole.css">
  <link rel="stylesheet" href="../../../../css/syntax.css">
  <link rel="stylesheet" href="../../../../css/lanyon.css">
  <link rel="stylesheet" href="../../../../css/idea.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=PT+Serif:400,400italic,700|PT+Sans:400">
  <link rel="stylesheet" href="../../../../fa/css/font-awesome.min.css">
  
  
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="../../../../assets/hippoPlus-icon.png">
  <link rel="shortcut icon" href="../../../../assets/hippoPlus-icon.png">

  
  <link rel="alternate" type="application/rss+xml" title="RSS" href="../../../../atom.xml">
</head>

  
  <body>

    
<input type="checkbox" class="sidebar-checkbox" id="sidebar-checkbox">


<div class="sidebar" id="sidebar">
  <div class="sidebar-item">
    <p>A place to gather useful information I keep on forgetting.</p>
  </div>

  <nav class="sidebar-nav">
    <a class="sidebar-nav-item " href="../../../../">Home</a>
    <a class="sidebar-nav-item " href="../../../../post">Posts</a>

    
    
      
        <a class="sidebar-nav-item " href="../../../../cpp/">CPP</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../emacs/">Emacs</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../expression/">Expression</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../latex/">LaTeX</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../markdown/">Markdown</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../math/stats/">Math/Stats</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../org-mode/">Org mode</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../python/">Python</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../r/">R</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../shell/">Shell</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../tools/">Tools</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../os/internet/">OS/Internet</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../about/">About</a>
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    

    
    
  </nav>

  <div class="sidebar-item">
  </div>
  
  <div class="sidebar-item">
    <div><a href="../../../../index.xml"><i class="fa fa-rss-square" aria-hidden="true"></i> RSS</a></div>
    <p>&copy; 2025 Jean Monlong. All rights reserved.</p>
  </div>
</div>


    
    <div class="wrap">
      <div class="masthead">
        <div class="container">
          <h3 class="masthead-title">
            <a href="../../../../" title="Home">Hippocamplus</a>
            <small>My Second Memory</small>
          </h3>
        </div>
      </div>

      <div class="container content">


<div class="post">
  <h1 class="post-title">First look at the gnomAD SV catalog</h1>
  <span class="post-date">Mar 31 2019
  
  
  
  
  
  <a href='../../../../tags/genome/' class='label'>genome</a>
  
  <a href='../../../../tags/sv/' class='label'>sv</a>
  
  
  
  </span>
  

<div id="TOC">
<ul>
<li><a href="#tldr">TL;DR</a></li>
<li><a href="#sv-types">SV types</a></li>
<li><a href="#detection-methods">Detection methods</a></li>
<li><a href="#allele-frequency">Allele frequency</a></li>
<li><a href="#size-distribution">Size distribution</a></li>
<li><a href="#sv-coordinates-confusion">SV coordinates confusion</a></li>
<li><a href="#potential-duplicates">Potential duplicates</a><ul>
<li><a href="#sv-calling-method">SV calling method</a></li>
<li><a href="#two-size-clusters-of-potentially-duplicated-complex-variants">Two size clusters of potentially duplicated complex variants</a></li>
<li><a href="#effect-on-allele-frequency-estimates">Effect on allele frequency estimates</a></li>
</ul></li>
<li><a href="#giab-comparison-to-investigate-sequence-resolution">GiaB comparison to investigate sequence resolution</a></li>
<li><a href="#chr2-and-end">CHR2 and END</a></li>
</ul>
</div>

<p><em>Edit April 1 2019: I added columns with the proportion of variants with PASS filter in the tables about the END/POS confusion and duplicated variants.</em></p>
<p><em>Edit September 23 2019: Some tables (SV types, methods), investigation of the inconsistent CHR2/END for a few SVs.</em></p>
<hr />
<p>The recent gnomAD-SV catalog is potentially a great resource that we could use to annotate SV calls or augment genome graphs with.
For now there is a <a href="https://www.biorxiv.org/content/10.1101/578674v1">preprint on bioRxiv</a> and a <a href="https://macarthurlab.org/2019/03/20/structural-variants-in-gnomad/">comprehensive blog post</a> that explains how the SVs were called and how to use the catalog.</p>
<p>I had a quick look at these variants and here are my observations/notes.
I’m using the following VCF file from the gnomAD download page: <a href="https://storage.googleapis.com/gnomad-public/papers/2019-sv/gnomad_v2_sv.sites.vcf.gz">gnomad_v2_sv.sites.vcf.gz</a>.</p>
<div id="tldr" class="section level2">
<h2>TL;DR</h2>
<ul>
<li>The vast majority of SVs look good and there is a lot of information in the VCF.</li>
<li>Some insertions and complex variants have contradicting POS/END fields (not sure what to do with those).</li>
<li>Thousands of SVs are so similar that they could be the same SVs that weren’t merged.</li>
<li>In a few cases, duplicates might lead to under-estimating the allele frequency.</li>
<li>Take-home message: don’t annotate SVs using only the best matching SV in gnomAD.</li>
<li>Most likely, the majority of variants are not really sequence-resolved (like in most SV catalogs).</li>
</ul>
</div>
<div id="sv-types" class="section level2">
<h2>SV types</h2>
<table>
<thead>
<tr class="header">
<th align="left">SVTYPE</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">DEL</td>
<td align="right">199,498</td>
</tr>
<tr class="even">
<td align="left">INS</td>
<td align="right">115,407</td>
</tr>
<tr class="odd">
<td align="left">BND</td>
<td align="right">72,411</td>
</tr>
<tr class="even">
<td align="left">DUP</td>
<td align="right">51,428</td>
</tr>
<tr class="odd">
<td align="left">CPX</td>
<td align="right">5,249</td>
</tr>
<tr class="even">
<td align="left">MCNV</td>
<td align="right">1,148</td>
</tr>
<tr class="odd">
<td align="left">INV</td>
<td align="right">707</td>
</tr>
<tr class="even">
<td align="left">CTX</td>
<td align="right">9</td>
</tr>
</tbody>
</table>
</div>
<div id="detection-methods" class="section level2">
<h2>Detection methods</h2>
<p>How many variants are supported by each method, for each SV type:</p>
<table>
<thead>
<tr class="header">
<th align="left">method</th>
<th align="right">BND</th>
<th align="right">CPX</th>
<th align="right">CTX</th>
<th align="right">DEL</th>
<th align="right">DUP</th>
<th align="right">INS</th>
<th align="right">INV</th>
<th align="right">MCNV</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">delly</td>
<td align="right">46,210</td>
<td align="right">5,050</td>
<td align="right">0</td>
<td align="right">122,492</td>
<td align="right">24,669</td>
<td align="right">2,360</td>
<td align="right">685</td>
<td align="right">185</td>
</tr>
<tr class="even">
<td align="left">depth</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">38,102</td>
<td align="right">21,607</td>
<td align="right">16</td>
<td align="right">0</td>
<td align="right">948</td>
</tr>
<tr class="odd">
<td align="left">manta</td>
<td align="right">31,510</td>
<td align="right">1,516</td>
<td align="right">9</td>
<td align="right">75,089</td>
<td align="right">22,809</td>
<td align="right">31,222</td>
<td align="right">181</td>
<td align="right">77</td>
</tr>
<tr class="even">
<td align="left">melt</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">17</td>
<td align="right">83,420</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>Note: the same variant is counted several times in this table if it is supported by multiple methods.</p>
<p>Interesting to see that MELT supports a few duplications (are they MEI close to an existing TE?) and that read depth changes support a few insertions (maybe some very recent MEI?).</p>
</div>
<div id="allele-frequency" class="section level2">
<h2>Allele frequency</h2>
<ul>
<li>AF field: sometimes one value (when biallelic), sometimes multiple values.</li>
<li>Only MCNVs contain multiple AF values (one for each copy number).</li>
<li>All MCNVs have a <em>CN=2</em> allele.</li>
</ul>
<p>I want one value of allele frequency per variant so I sum up the AF values across alleles excluding the <em>CN=2</em> allele for MCNVs.</p>
<p><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/af-1.png" width="960" /></p>
</div>
<div id="size-distribution" class="section level2">
<h2>Size distribution</h2>
<p><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/size-1.png" width="960" /><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/size-2.png" width="960" /><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/size-3.png" width="960" /><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/size-4.png" width="960" /></p>
<p>The <strong>peak at 5 kbp is a bit puzzling</strong>.
Maybe it’s a technical artifact, e.g. due to a method that detects <em>DEL</em>/<em>DUP</em> only down to 5 kbp.
Looking at the variants with absolute <em>SVLEN</em> of exactly 5 kbp:</p>
<table>
<thead>
<tr class="header">
<th align="left">methods</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">depth</td>
<td align="right">3,185</td>
</tr>
<tr class="even">
<td align="left">delly_depth</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">delly</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">delly_depth_manta</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">manta</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<p>It’s most likely an artifact of the <em>depth</em> algorithm.
Does it affect the allele frequency distribution?</p>
<p><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/meths-5kbp-af-1.png" width="960" /></p>
<p>The tail of the distribution is longer in the 5-6 kbp class.</p>
</div>
<div id="sv-coordinates-confusion" class="section level2">
<h2>SV coordinates confusion</h2>
<p>It seems like the <strong><em>END</em> value is not always larger than the position <em>POS</em></strong> in the VCF.
Most of it comes from <em>CTX</em> (reciprocal chromosomal translocation) variants or translocations which makes sense: the END might relate to the second breakpoint, potentially anywhere in the genome.</p>
<table>
<thead>
<tr class="header">
<th align="left">SVTYPE</th>
<th align="right">end.before.pos</th>
<th align="right">prop</th>
<th align="right">prop.PASS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">INS</td>
<td align="right">358</td>
<td align="right">0.0031</td>
<td align="right">0.9973</td>
</tr>
<tr class="even">
<td align="left">CPX</td>
<td align="right">289</td>
<td align="right">0.0551</td>
<td align="right">0.9737</td>
</tr>
<tr class="odd">
<td align="left">BND</td>
<td align="right">44</td>
<td align="right">0.0006</td>
<td align="right">0.0000</td>
</tr>
<tr class="even">
<td align="left">CTX</td>
<td align="right">6</td>
<td align="right">0.6667</td>
<td align="right">1.0000</td>
</tr>
<tr class="odd">
<td align="left">DEL</td>
<td align="right">0</td>
<td align="right">0.0000</td>
<td align="right">0.8833</td>
</tr>
<tr class="even">
<td align="left">DUP</td>
<td align="right">0</td>
<td align="right">0.0000</td>
<td align="right">0.9772</td>
</tr>
<tr class="odd">
<td align="left">INV</td>
<td align="right">0</td>
<td align="right">0.0000</td>
<td align="right">0.9986</td>
</tr>
<tr class="even">
<td align="left">MCNV</td>
<td align="right">0</td>
<td align="right">0.0000</td>
<td align="right">0.0000</td>
</tr>
</tbody>
</table>
<p>Some are insertions and I’m not sure how to interpret them.
The vast majority of these variants have a <em>PASS</em> filter.</p>
<p><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/ins-end-1.png" width="960" /></p>
<p>For most of these insertions the <em>END</em> is not that far from the <em>POS</em> (&lt;100bp) but <strong>for some the <em>END</em> and <em>POS</em> are hundreds of kbp apart</strong>.
Hum.
Let’s have a look at some of those:</p>
<pre><code>## GRanges object with 3 ranges and 13 metadata columns:
##                         seqnames                 ranges strand |
##                            &lt;Rle&gt;              &lt;IRanges&gt;  &lt;Rle&gt; |
##   gnomAD_v2_INS_4_34681        4 [116015282, 116015282]      * |
##   gnomAD_v2_INS_2_13594        2 [ 85885163,  85885163]      * |
##   gnomAD_v2_INS_9_69619        9 [ 28199346,  28199346]      * |
##                         paramRangeID            REF             ALT
##                             &lt;factor&gt; &lt;DNAStringSet&gt; &lt;CharacterList&gt;
##   gnomAD_v2_INS_4_34681         &lt;NA&gt;              N    &lt;INS:ME:ALU&gt;
##   gnomAD_v2_INS_2_13594         &lt;NA&gt;              N    &lt;INS:ME:ALU&gt;
##   gnomAD_v2_INS_9_69619         &lt;NA&gt;              N           &lt;INS&gt;
##                              QUAL      FILTER      SVTYPE     SVLEN
##                         &lt;numeric&gt; &lt;character&gt; &lt;character&gt; &lt;integer&gt;
##   gnomAD_v2_INS_4_34681       646        PASS         INS       281
##   gnomAD_v2_INS_2_13594       652        PASS         INS       279
##   gnomAD_v2_INS_9_69619       407        PASS         INS       136
##                               END        CHR2    CPX_TYPE          methods
##                         &lt;integer&gt; &lt;character&gt; &lt;character&gt;      &lt;character&gt;
##   gnomAD_v2_INS_4_34681  44565090           4        &lt;NA&gt; delly_manta_melt
##   gnomAD_v2_INS_2_13594   8372664           2        &lt;NA&gt; delly_manta_melt
##   gnomAD_v2_INS_9_69619  17898584           9        &lt;NA&gt;      delly_manta
##                                            ID        AF
##                                   &lt;character&gt; &lt;numeric&gt;
##   gnomAD_v2_INS_4_34681 gnomAD_v2_INS_4_34681  0.001490
##   gnomAD_v2_INS_2_13594 gnomAD_v2_INS_2_13594  0.005355
##   gnomAD_v2_INS_9_69619 gnomAD_v2_INS_9_69619  0.057464
##   -------
##   seqinfo: 24 sequences from an unspecified genome</code></pre>
<p>I don’t see anything else suspicious.</p>
<p>For complex variants, the <em>POS</em> and <em>END</em> are always less than 100 bp apart and all for one type of complex SV: <strong>“dispersed duplications”</strong>.
If I understand correctly, this type of complex SV is pretty much an insertion of a sequence already in the genome (but not a transposable element I guess?).</p>
<p><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/cpx-end-1.png" width="960" /></p>
<p>I’m not sure how to interpret them either.</p>
<p>For now, I’ll use the <strong>POS field for both insertions and dispersed duplications</strong> when manipulating the variants as genomic intervals.
So I’ll assume that they are both simple insertions (which I think is fair enough).</p>
</div>
<div id="potential-duplicates" class="section level2">
<h2>Potential duplicates</h2>
<p>Are there any pairs of variants with a <strong>reciprocal overlap higher than 90%</strong>?
For insertions, I match them if they are located at less than 30 bp from each other and their size is 90% similar.</p>
<p>Note: Here I don’t consider <em>BND</em> or <em>CTX</em> variants, and I count each pair of variants only once.</p>
<table>
<thead>
<tr class="header">
<th align="left">types</th>
<th align="right">n</th>
<th align="right">prop.both.PASS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">DEL_DEL</td>
<td align="right">3,335</td>
<td align="right">0.8237</td>
</tr>
<tr class="even">
<td align="left">INS_INS</td>
<td align="right">2,221</td>
<td align="right">0.9955</td>
</tr>
<tr class="odd">
<td align="left">DUP_DUP</td>
<td align="right">2,057</td>
<td align="right">0.9728</td>
</tr>
<tr class="even">
<td align="left">CPX_CPX</td>
<td align="right">1,077</td>
<td align="right">0.0529</td>
</tr>
<tr class="odd">
<td align="left">DEL_DUP</td>
<td align="right">797</td>
<td align="right">0.8821</td>
</tr>
<tr class="even">
<td align="left">CPX_DEL</td>
<td align="right">253</td>
<td align="right">0.9921</td>
</tr>
<tr class="odd">
<td align="left">DEL_MCNV</td>
<td align="right">117</td>
<td align="right">0.0000</td>
</tr>
<tr class="even">
<td align="left">CPX_DUP</td>
<td align="right">66</td>
<td align="right">0.9091</td>
</tr>
<tr class="odd">
<td align="left">DUP_MCNV</td>
<td align="right">50</td>
<td align="right">0.0000</td>
</tr>
<tr class="even">
<td align="left">DEL_INV</td>
<td align="right">10</td>
<td align="right">1.0000</td>
</tr>
<tr class="odd">
<td align="left">CPX_INS</td>
<td align="right">6</td>
<td align="right">1.0000</td>
</tr>
<tr class="even">
<td align="left">DUP_INV</td>
<td align="right">2</td>
<td align="right">1.0000</td>
</tr>
<tr class="odd">
<td align="left">CPX_INV</td>
<td align="right">1</td>
<td align="right">1.0000</td>
</tr>
<tr class="even">
<td align="left">INV_INV</td>
<td align="right">1</td>
<td align="right">1.0000</td>
</tr>
</tbody>
</table>
<p>There are <strong>thousands of deletions, insertions and duplications with high reciprocal overlap.</strong>
Also many complex variants but I can imagine that it’s difficult to merge those.
For the vast majority of pairs, both variants have a <em>PASS</em> filter.
However, for most of the complex variants duplicates, one of the variant doesn’t have a PASS filter.
<strong>Using <em>FILTER==PASS</em> would remove almost all the complex SV duplicates.</strong></p>
<p>Let’s look at the size distribution of these potential duplicates.</p>
<p><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/dups-size-1.png" width="960" /></p>
<p>Wow, very “peaky”.
Let’s zoom in with no log-scale:</p>
<p><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/dups-peaks-1.png" width="960" /><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/dups-peaks-2.png" width="960" /><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/dups-peaks-3.png" width="960" /></p>
<ul>
<li>The “Alu” peak around 300 bp is made of around 200 insertion pairs.</li>
<li>The “L1” peak around 6 kbp is made of around 30 insertion pairs and almost as many duplication pairs.</li>
<li>Again the technical peak at 5 kbp, here mostly pairs of deletions.</li>
<li>The complex variants that are potentially duplicated form two size clusters.</li>
</ul>
<div id="sv-calling-method" class="section level3">
<h3>SV calling method</h3>
<p>Duplicates can occur when merging two sets of variants, for example the call sets of different methods.
<strong>Are the potential duplicates from different methods?</strong></p>
<p>Top 10 of the most common pairs:</p>
<table>
<thead>
<tr class="header">
<th align="left">types</th>
<th align="left">methods</th>
<th align="right">n</th>
<th align="right">prop.both.PASS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">INS_INS</td>
<td align="left">manta__manta</td>
<td align="right">1,639</td>
<td align="right">0.9969</td>
</tr>
<tr class="even">
<td align="left">DEL_DEL</td>
<td align="left">delly__delly</td>
<td align="right">1,110</td>
<td align="right">0.6009</td>
</tr>
<tr class="odd">
<td align="left">DEL_DEL</td>
<td align="left">manta__manta</td>
<td align="right">652</td>
<td align="right">0.9417</td>
</tr>
<tr class="even">
<td align="left">DUP_DUP</td>
<td align="left">delly__delly</td>
<td align="right">576</td>
<td align="right">0.9688</td>
</tr>
<tr class="odd">
<td align="left">CPX_CPX</td>
<td align="left">delly__delly</td>
<td align="right">528</td>
<td align="right">0.0436</td>
</tr>
<tr class="even">
<td align="left">INS_INS</td>
<td align="left">melt__melt</td>
<td align="right">489</td>
<td align="right">0.9918</td>
</tr>
<tr class="odd">
<td align="left">DUP_DUP</td>
<td align="left">manta__manta</td>
<td align="right">443</td>
<td align="right">0.9842</td>
</tr>
<tr class="even">
<td align="left">DEL_DUP</td>
<td align="left">depth__depth</td>
<td align="right">395</td>
<td align="right">0.8785</td>
</tr>
<tr class="odd">
<td align="left">DUP_DUP</td>
<td align="left">delly__manta</td>
<td align="right">376</td>
<td align="right">0.9787</td>
</tr>
<tr class="even">
<td align="left">DEL_DEL</td>
<td align="left">delly__manta</td>
<td align="right">319</td>
<td align="right">0.9310</td>
</tr>
</tbody>
</table>
<p><strong>For thousands of potential duplicates the variants were called by the same algorithm.</strong>
The question is then: were these methods run once in a “single-sample” mode or on all the samples together?
I think at least Delly and MELT can do the genotyping step across all samples together.
Even in those, there can be duplicates in the discovery phase that are not merged properly before genotyping.</p>
<p>I’m still not sure if these deletions/duplications/MEIs are actually different SVs.
Especially the mobile element insertions, I would lean toward them being duplicates.
Otherwise, it means the TE jumped and then got an indel later.
It’s possible but maybe not that often.
Also, the comparison with GiaB below shows that we are not there yet in term of bp resolution.
<strong>It might be safer to assume these variants are duplicates.</strong></p>
</div>
<div id="two-size-clusters-of-potentially-duplicated-complex-variants" class="section level3">
<h3>Two size clusters of potentially duplicated complex variants</h3>
<p>The vast majority of the complex SV duplicates would disappear if we were filtering variants with <em>FILTER=PASS</em>.
I’m still curious about the fact that two groups of potentially duplicated complex variants clustered by size.
Defining size clusters 1 and 2 for the complex variants of sizes around 120 kbp and 160 kbp, respectively.</p>
<p><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/dups-cpx-1.png" width="960" /></p>
<p>Some do cluster together in the genome and, looking up a few of them, they seem to be in segmental duplications.</p>
<table>
<thead>
<tr class="header">
<th align="left">chr</th>
<th align="left">types</th>
<th align="left">methods</th>
<th align="left">coord</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">7</td>
<td align="left">CPX_CPX</td>
<td align="left">delly__delly</td>
<td align="left"><a href="https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&amp;position=chr7:64296630-64416847">7:64296630-64416847</a></td>
</tr>
<tr class="even">
<td align="left">11</td>
<td align="left">CPX_CPX</td>
<td align="left">delly__delly</td>
<td align="left"><a href="https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&amp;position=chr11:4253841-4360790">11:4253841-4360790</a></td>
</tr>
<tr class="odd">
<td align="left">12</td>
<td align="left">CPX_CPX</td>
<td align="left">delly__delly</td>
<td align="left"><a href="https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&amp;position=chr12:9556527-9723033">12:9556527-9723033</a></td>
</tr>
<tr class="even">
<td align="left">14</td>
<td align="left">CPX_CPX</td>
<td align="left">delly__delly</td>
<td align="left"><a href="https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&amp;position=chr14:106040016-106164465">14:106040016-106164465</a></td>
</tr>
<tr class="odd">
<td align="left">16</td>
<td align="left">CPX_CPX</td>
<td align="left">delly__delly_manta</td>
<td align="left"><a href="https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&amp;position=chr16:33350710-33513016">16:33350710-33513016</a></td>
</tr>
</tbody>
</table>
</div>
<div id="effect-on-allele-frequency-estimates" class="section level3">
<h3>Effect on allele frequency estimates</h3>
<p><strong>What is the potential effect of these duplicated variants on the frequency annotation?</strong></p>
<p>Below, I compare the allele frequencies of each variant and its potential duplicate.</p>
<p><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/dups-af-1.png" width="960" /></p>
<p><strong>Most pairs of potential duplicates have similar frequencies.</strong>
Does that mean that the frequencies of these variants are supposed to be the double of what we have?</p>
<p>Let’s zoom in to the long tail with the highest difference in allele frequency.
To make the graph clearer, I compare the lowest frequency to the highest.</p>
<p><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/dups-af-scatter-1.png" width="960" /></p>
<ul>
<li>Some SVs with low frequencies might be a duplicate of a SV with higher frequency.</li>
<li>E.g. 827 of these variants have AF&lt;0.1% while the other variant has AF&gt;1%.</li>
</ul>
<p>Although it’s only a minority of cases, <strong>we should be careful when matching our SVs to this catalog.</strong>
Otherwise we might think that a variant is rarer than it is.
Instead of annotating a SV with the allele frequency of the most similar SV in gnomAD-SV, a more robust strategy would be to use the maximum frequency across all SVs that are similar enough.
That’s what we had done when annotating CNVs in our <a href="https://doi.org/10.1371/journal.pgen.1007285">epilepsy study</a>, although we were being maybe too conservative by considering any overlapping SV.
For short deletions it’s fine, but in general something like <em>reciprocal overlap &gt; XX%</em> would be less conservative.</p>
</div>
</div>
<div id="giab-comparison-to-investigate-sequence-resolution" class="section level2">
<h2>GiaB comparison to investigate sequence resolution</h2>
<p>The Genome in a Bottle consortium combined many types of techonologies (ultra deep short-read, linked-reads, long reads, optical mapping) to generate a <strong>high-quality SV catalog</strong>.
Maybe not all variants are correct but it’s one of the best quality catalog we have.
I want to quickly compare these SVs with the ones in the gnomAD catalog.
If many variants look exactly the same in both datasets, it would be good evidence that they are both sequence-resolved.</p>
<p>I downloaded the <a href="ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/analysis/NIST_SVs_Integration_v0.6/">NIST SVs Integration v0.6</a>.
This catalog also uses GRCh37.
It includes short indels that I decided to filter out, keeping only variants of size 50 bp or more.</p>
<table>
<thead>
<tr class="header">
<th align="left">SVTYPE</th>
<th align="right">n</th>
<th align="right">mean.bp</th>
<th align="right">min.bp</th>
<th align="right">max.bp</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">DEL</td>
<td align="right">14,588</td>
<td align="right">1,715</td>
<td align="right">50</td>
<td align="right">997,115</td>
</tr>
<tr class="even">
<td align="left">INS</td>
<td align="right">15,432</td>
<td align="right">595</td>
<td align="right">50</td>
<td align="right">125,187</td>
</tr>
</tbody>
</table>
<p>I’m curious to compare the exact breakpoint locations of SVs shared by both catalogs.
Do they have exactly the same breakpoints/sequence?
To make things easier let’s just look at deletions.</p>
<p><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/giab-del-1.png" width="960" /></p>
<p>Many deletions in the gnomAD catalog are <strong>extremely similar</strong> to the GiaB dataset <strong>but not exactly the same</strong>.
Some examples (randomly selected):</p>
<table>
<thead>
<tr class="header">
<th align="left">seqnames</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="left">SVTYPE</th>
<th align="right">SVLEN</th>
<th align="left">catalog</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">7</td>
<td align="right">93541818</td>
<td align="right">93542540</td>
<td align="left">DEL</td>
<td align="right">720</td>
<td align="left">GiaB</td>
</tr>
<tr class="even">
<td align="left">7</td>
<td align="right">93541816</td>
<td align="right">93542539</td>
<td align="left">DEL</td>
<td align="right">723</td>
<td align="left">gnomAD</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="right">34523655</td>
<td align="right">34525720</td>
<td align="left">DEL</td>
<td align="right">2065</td>
<td align="left">GiaB</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="right">34523656</td>
<td align="right">34525719</td>
<td align="left">DEL</td>
<td align="right">2063</td>
<td align="left">gnomAD</td>
</tr>
<tr class="odd">
<td align="left">X</td>
<td align="right">70424308</td>
<td align="right">70430224</td>
<td align="left">DEL</td>
<td align="right">5916</td>
<td align="left">GiaB</td>
</tr>
<tr class="even">
<td align="left">X</td>
<td align="right">70424367</td>
<td align="right">70430283</td>
<td align="left">DEL</td>
<td align="right">5916</td>
<td align="left">gnomAD</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="right">96500161</td>
<td align="right">96503043</td>
<td align="left">DEL</td>
<td align="right">2882</td>
<td align="left">GiaB</td>
</tr>
<tr class="even">
<td align="left">9</td>
<td align="right">96500163</td>
<td align="right">96503042</td>
<td align="left">DEL</td>
<td align="right">2879</td>
<td align="left">gnomAD</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="right">83125959</td>
<td align="right">83127570</td>
<td align="left">DEL</td>
<td align="right">1611</td>
<td align="left">GiaB</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="right">83125976</td>
<td align="right">83127569</td>
<td align="left">DEL</td>
<td align="right">1593</td>
<td align="left">gnomAD</td>
</tr>
<tr class="odd">
<td align="left">12</td>
<td align="right">69255776</td>
<td align="right">69256518</td>
<td align="left">DEL</td>
<td align="right">742</td>
<td align="left">GiaB</td>
</tr>
<tr class="even">
<td align="left">12</td>
<td align="right">69255779</td>
<td align="right">69256517</td>
<td align="left">DEL</td>
<td align="right">738</td>
<td align="left">gnomAD</td>
</tr>
</tbody>
</table>
<p>This quick comparison suggests that the variants are not really sequence-resolved.
<strong>In gnomAD, the GiaB dataset, or both, the breakpoints might have a few errors.</strong>
That will be important when genotyping these SVs or injecting them in genome graphs.</p>
<p>Of note, I’m not sure if the variants were left-aligned.
In this case it wouldn’t make a big difference because we see that the deletions are discordant in term of size also.
Still, ideally both VCF should be normalized first to better estimate how many deletions are exactly matched.</p>
</div>
<div id="chr2-and-end" class="section level2">
<h2>CHR2 and END</h2>
<p><em>CHR2</em> is supposed to be the “Chromosome for END coordinate”.
However I noticed that it might not be the case sometimes.
For example, when <em>CHR2</em> is different from <em>#CHROM</em> but the <em>END</em> is very close to the variant’s position.
In those cases it might be that the <em>END</em> was set to the position in the <em>#CHROM</em>/<em>POS</em> side, like for insertions for example.</p>
<p><img src="../../../../post/2019-03-31-gnomadSV_files/figure-html/chr2end-1.png" width="960" /></p>
<table>
<thead>
<tr class="header">
<th align="left">filter.pass</th>
<th align="left">SVTYPE</th>
<th align="right">pos.equal.end</th>
<th align="right">pos.20bp.end</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">FALSE</td>
<td align="left">BND</td>
<td align="right">21,079</td>
<td align="right">21,079</td>
</tr>
<tr class="even">
<td align="left">FALSE</td>
<td align="left">CPX</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">FALSE</td>
<td align="left">INS</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">TRUE</td>
<td align="left">CPX</td>
<td align="right">132</td>
<td align="right">132</td>
</tr>
<tr class="odd">
<td align="left">TRUE</td>
<td align="left">INS</td>
<td align="right">75</td>
<td align="right">75</td>
</tr>
<tr class="even">
<td align="left">TRUE</td>
<td align="left">CTX</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>It happens a lot (the y-axis is log-scale) but mostly in variants without a <em>FILTER=PASS</em>.
Just a few dozens variants with <em>FILTER=PASS</em> may have a problem with their <em>CHR2</em> or <em>END</em>.</p>
<p>Some examples:</p>
<pre><code>##   seqnames    start      end width strand paramRangeID REF   ALT QUAL
## 1        1  8620717  8620717     1      *         &lt;NA&gt;   N &lt;INS&gt;  999
## 2        1  9121444  9121444     1      *         &lt;NA&gt;   N &lt;CPX&gt;  331
## 3        1 23636591 23636591     1      *         &lt;NA&gt;   N &lt;CPX&gt;  330
## 4        1 30893626 30893626     1      *         &lt;NA&gt;   N &lt;INS&gt;  534
## 5        1 59880893 59880893     1      *         &lt;NA&gt;   N &lt;INS&gt;  351
## 6        1 66188203 66188203     1      *         &lt;NA&gt;   N &lt;INS&gt;  175
##   FILTER SVTYPE SVLEN      END CHR2 CPX_TYPE methods                   ID
## 1   PASS    INS   277  8620717   22     &lt;NA&gt;   manta  gnomAD_v2_INS_1_295
## 2   PASS    CPX   718  9121444   14     dDUP   manta   gnomAD_v2_CPX_1_20
## 3   PASS    CPX   270 23636591    2     dDUP   manta   gnomAD_v2_CPX_1_57
## 4   PASS    INS   172 30893626    X     &lt;NA&gt;   manta gnomAD_v2_INS_1_1114
## 5   PASS    INS   340 59880893    2     &lt;NA&gt;   manta gnomAD_v2_INS_1_2293
## 6   PASS    INS   344 66188203    7     &lt;NA&gt;   manta gnomAD_v2_INS_1_2601
##         AF
## 1 0.000047
## 2 0.225787
## 3 0.000047
## 4 0.001762
## 5 0.001261
## 6 0.000047</code></pre>
</div>

</div>

      </div>
    </div>

    

    <label for="sidebar-checkbox" class="sidebar-toggle"></label>

    <script src="../../../../assets/highlight.pack.js"></script>
    <script src="../../../../assets/r.min.js"></script>
    
    <script>
      hljs.configure({languages: ['r']});
      hljs.initHighlightingOnLoad();
    </script>

    <script src="//yihui.name/js/math-code.js"></script>
    <script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
    
  </body>
</html>

