<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      MUMmerplots with ggplot2 &middot; Hippocamplus
    
  </title>

  
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-113785126-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    
    gtag('config', "UA-113785126-1");
  </script>
  
  
  <link rel="stylesheet" href="../../../../css/poole.css">
  <link rel="stylesheet" href="../../../../css/syntax.css">
  <link rel="stylesheet" href="../../../../css/lanyon.css">
  <link rel="stylesheet" href="../../../../css/idea.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=PT+Serif:400,400italic,700|PT+Sans:400">
  <link rel="stylesheet" href="../../../../fa/css/font-awesome.min.css">
  
  
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="../../../../assets/hippoPlus-icon.png">
  <link rel="shortcut icon" href="../../../../assets/hippoPlus-icon.png">

  
  <link rel="alternate" type="application/rss+xml" title="RSS" href="../../../../atom.xml">
</head>

  
  <body>

    
<input type="checkbox" class="sidebar-checkbox" id="sidebar-checkbox">


<div class="sidebar" id="sidebar">
  <div class="sidebar-item">
    <p>A place to gather useful information I keep on forgetting.</p>
  </div>

  <nav class="sidebar-nav">
    <a class="sidebar-nav-item " href="../../../../">Home</a>
    <a class="sidebar-nav-item " href="../../../../post">Posts</a>

    
    
      
        <a class="sidebar-nav-item " href="../../../../cpp/">CPP</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../emacs/">Emacs</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../expression/">Expression</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../latex/">LaTeX</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../markdown/">Markdown</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../math/stats/">Math/Stats</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../org-mode/">Org mode</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../python/">Python</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../r/">R</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../shell/">Shell</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../tools/">Tools</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../os/internet/">OS/Internet</a>
      
    
      
        <a class="sidebar-nav-item " href="../../../../about/">About</a>
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    
      
    

    
    
  </nav>

  <div class="sidebar-item">
  </div>
  
  <div class="sidebar-item">
    <div><a href="../../../../index.xml"><i class="fa fa-rss-square" aria-hidden="true"></i> RSS</a></div>
    <div><a href="https://twitter.com/JMonlong"><i class="fa fa-twitter" aria-hidden="true"></i> @JMonlong</a></div>
    <p>&copy; 2020 Jean Monlong. All rights reserved.</p>
  </div>
</div>


    
    <div class="wrap">
      <div class="masthead">
        <div class="container">
          <h3 class="masthead-title">
            <a href="../../../../" title="Home">Hippocamplus</a>
            <small>My Second Memory</small>
          </h3>
        </div>
      </div>

      <div class="container content">


<div class="post">
  <h1 class="post-title">MUMmerplots with ggplot2</h1>
  <span class="post-date">Sep 19 2017
  
  
  
  
  
  <a href='../../../../tags/r/' class='label'>R</a>
  
  <a href='../../../../tags/plot/' class='label'>plot</a>
  
  
  
  </span>
  


<p><em>Update Oct 28 2018: added reference id (rid) to be able to visualize multiple reference regions. Also uploaded the example data somewhere.</em></p>
<pre class="r"><code>library(dplyr)
library(magrittr)
library(GenomicRanges)
library(knitr)
library(ggplot2)
library(tidyr)</code></pre>
<div id="mummer-plot" class="section level2">
<h2>MUMmer plot</h2>
<p>The MUMmer plot that I want to reproduce showed three contigs overlapping a region of chr 14. I had filtered the delta file with <code>delta-filter -l 10000 -q -r</code> to get only the contigs with the best alignments. I had used <code>mummerplot</code> with the <code>-l</code> layout option to reorder and orient the sequences to have a nice diagonal.</p>
<div class="figure">
<img src="../../../../imgs/mumplot-example.png" />

</div>
</div>
<div id="delta-file" class="section level2">
<h2>Delta file</h2>
<p>The delta file is the default output of the <a href="http://mummer.sourceforge.net/manual/#nucmer">NUCmer alignment script</a>. The format of the delta file is described more <a href="http://mummer.sourceforge.net/manual/#nucmeroutput">here</a>.</p>
<p>The delta file used in this post can be downloaded <a href="https://www.dropbox.com/s/3zscsbbex6rgemo/mumplot-example.delta?dl=0">here</a>. Otherwise, in R:</p>
<pre class="r"><code>if(!file.exists(&#39;mumplot-example.delta&#39;)){
    download.file(&#39;https://dl.dropboxusercontent.com/s/3zscsbbex6rgemo/mumplot-example.delta?dl0&#39;,
                  &#39;mumplot-example.delta&#39;)
}</code></pre>
</div>
<div id="read-a-delta-file" class="section level2">
<h2>Read a delta file</h2>
<pre class="r"><code>readDelta &lt;- function(deltafile){
  lines = scan(deltafile, &#39;a&#39;, sep=&#39;\n&#39;, quiet=TRUE)
  lines = lines[-1]
  lines.l = strsplit(lines, &#39; &#39;)
  lines.len = lapply(lines.l, length) %&gt;% as.numeric
  lines.l = lines.l[lines.len != 1]
  lines.len = lines.len[lines.len != 1]
  head.pos = which(lines.len == 4)
  head.id = rep(head.pos, c(head.pos[-1], length(lines.l)+1)-head.pos)
  mat = matrix(as.numeric(unlist(lines.l[lines.len==7])), 7)
  res = as.data.frame(t(mat[1:5,]))
  colnames(res) = c(&#39;rs&#39;,&#39;re&#39;,&#39;qs&#39;,&#39;qe&#39;,&#39;error&#39;)
  res$qid = unlist(lapply(lines.l[head.id[lines.len==7]], &#39;[&#39;, 2))
  res$rid = unlist(lapply(lines.l[head.id[lines.len==7]], &#39;[&#39;, 1)) %&gt;% gsub(&#39;^&gt;&#39;, &#39;&#39;, .)
  res$strand = ifelse(res$qe-res$qs &gt; 0, &#39;+&#39;, &#39;-&#39;)
  res
}

mumgp = readDelta(&quot;mumplot-example.delta&quot;)

mumgp %&gt;% head %&gt;% kable</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">rs</th>
<th align="right">re</th>
<th align="right">qs</th>
<th align="right">qe</th>
<th align="right">error</th>
<th align="left">qid</th>
<th align="left">rid</th>
<th align="left">strand</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">265577</td>
<td align="right">265842</td>
<td align="right">108520</td>
<td align="right">108254</td>
<td align="right">46</td>
<td align="left">Contig0</td>
<td align="left">chr14:105095800-107043718</td>
<td align="left">-</td>
</tr>
<tr class="even">
<td align="right">265577</td>
<td align="right">265842</td>
<td align="right">106438</td>
<td align="right">106172</td>
<td align="right">46</td>
<td align="left">Contig0</td>
<td align="left">chr14:105095800-107043718</td>
<td align="left">-</td>
</tr>
<tr class="odd">
<td align="right">306695</td>
<td align="right">306968</td>
<td align="right">138241</td>
<td align="right">138515</td>
<td align="right">31</td>
<td align="left">Contig0</td>
<td align="left">chr14:105095800-107043718</td>
<td align="left">+</td>
</tr>
<tr class="even">
<td align="right">1016956</td>
<td align="right">1017364</td>
<td align="right">27806</td>
<td align="right">27394</td>
<td align="right">62</td>
<td align="left">Contig0</td>
<td align="left">chr14:105095800-107043718</td>
<td align="left">-</td>
</tr>
<tr class="odd">
<td align="right">1723715</td>
<td align="right">1723990</td>
<td align="right">34123</td>
<td align="right">33845</td>
<td align="right">26</td>
<td align="left">Contig0</td>
<td align="left">chr14:105095800-107043718</td>
<td align="left">-</td>
</tr>
<tr class="even">
<td align="right">1767531</td>
<td align="right">1767813</td>
<td align="right">33842</td>
<td align="right">34123</td>
<td align="right">24</td>
<td align="left">Contig0</td>
<td align="left">chr14:105095800-107043718</td>
<td align="left">+</td>
</tr>
</tbody>
</table>
</div>
<div id="filter-contigs-with-poor-alignments" class="section level2">
<h2>Filter contigs with poor alignments</h2>
<p>For now, I filter contigs simply based on the size of the aligned segment. I keep only contigs with at least one aligned segment larger than a minimum size. Smaller alignment in these contigs are kept if in the same range as the large aligned segments. Eventually, I could also filter segment based on the number/proportion of errors.</p>
<pre class="r"><code>filterMum &lt;- function(df, minl=1000, flanks=1e4){
    coord = df %&gt;% filter(abs(re-rs)&gt;minl) %&gt;% group_by(qid, rid) %&gt;%
        summarize(qsL=min(qs)-flanks, qeL=max(qe)+flanks, rs=median(rs)) %&gt;%
        ungroup %&gt;% arrange(desc(rs)) %&gt;%
        mutate(qid=factor(qid, levels=unique(qid))) %&gt;% select(-rs)
    merge(df, coord) %&gt;% filter(qs&gt;qsL, qe&lt;qeL) %&gt;%
        mutate(qid=factor(qid, levels=levels(coord$qid))) %&gt;% select(-qsL, -qeL)
}

mumgp.filt = filterMum(mumgp, minl=1e4)
mumgp.filt %&gt;% head %&gt;% kable</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">qid</th>
<th align="left">rid</th>
<th align="right">rs</th>
<th align="right">re</th>
<th align="right">qs</th>
<th align="right">qe</th>
<th align="right">error</th>
<th align="left">strand</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Contig1475</td>
<td align="left">chr14:105095800-107043718</td>
<td align="right">1663946</td>
<td align="right">1665485</td>
<td align="right">331648</td>
<td align="right">330113</td>
<td align="right">171</td>
<td align="left">-</td>
</tr>
<tr class="even">
<td align="left">Contig1475</td>
<td align="left">chr14:105095800-107043718</td>
<td align="right">1662200</td>
<td align="right">1684396</td>
<td align="right">126037</td>
<td align="right">103837</td>
<td align="right">234</td>
<td align="left">-</td>
</tr>
<tr class="odd">
<td align="left">Contig1475</td>
<td align="left">chr14:105095800-107043718</td>
<td align="right">1581333</td>
<td align="right">1582738</td>
<td align="right">244635</td>
<td align="right">243233</td>
<td align="right">87</td>
<td align="left">-</td>
</tr>
<tr class="even">
<td align="left">Contig1475</td>
<td align="left">chr14:105095800-107043718</td>
<td align="right">1597381</td>
<td align="right">1610746</td>
<td align="right">145948</td>
<td align="right">132626</td>
<td align="right">157</td>
<td align="left">-</td>
</tr>
<tr class="odd">
<td align="left">Contig1475</td>
<td align="left">chr14:105095800-107043718</td>
<td align="right">1610278</td>
<td align="right">1623358</td>
<td align="right">130561</td>
<td align="right">117468</td>
<td align="right">200</td>
<td align="left">-</td>
</tr>
<tr class="even">
<td align="left">Contig1475</td>
<td align="left">chr14:105095800-107043718</td>
<td align="right">1616542</td>
<td align="right">1618080</td>
<td align="right">331648</td>
<td align="right">330113</td>
<td align="right">146</td>
<td align="left">-</td>
</tr>
</tbody>
</table>
</div>
<div id="graph" class="section level2">
<h2>Graph</h2>
<p>I’m going for the same style as <code>mummerplot</code> to compare.</p>
<pre class="r"><code>ggplot(mumgp.filt, aes(x=rs, xend=re, y=qs, yend=qe, colour=strand)) + geom_segment() +
    geom_point(alpha=.5) + facet_grid(qid~., scales=&#39;free&#39;, space=&#39;free&#39;, switch=&#39;y&#39;) +
    theme_bw() + theme(strip.text.y=element_text(angle=180, size=5),
                       legend.position=c(.99,.01), legend.justification=c(1,0),
                       strip.background=element_blank(),
                       axis.text.y=element_blank(), axis.ticks.y=element_blank()) +
    xlab(&#39;reference sequence&#39;) + ylab(&#39;assembly&#39;) + scale_colour_brewer(palette=&#39;Set1&#39;)</code></pre>
<p><img src="../../../../post/2017-09-19-MumplotsWithGgplot2_files/figure-html/graph-1.png" width="960" /></p>
<p>Not bad but it would look nicer if we flipped the contigs to have more or less a diagonal.</p>
</div>
<div id="diagonalize" class="section level2">
<h2>Diagonalize</h2>
<p>For each contig, I compute the major strand (strand with most bases aligned) and flip if necessary. The contigs are also ordered based on the reference region with most bases and the weighted means of the start position in this matched reference region.</p>
<pre class="r"><code>diagMum &lt;- function(df){
    ## Find best qid order
    rid.o = df %&gt;% group_by(qid, rid) %&gt;% summarize(base=sum(abs(qe-qs)),
                                                    rs=weighted.mean(rs, abs(qe-qs))) %&gt;%
        ungroup %&gt;% arrange(desc(base)) %&gt;% group_by(qid) %&gt;% do(head(., 1)) %&gt;%
        ungroup %&gt;% arrange(desc(rid), desc(rs)) %&gt;%
        mutate(qid=factor(qid, levels=unique(qid)))
    ## Find best qid strand
    major.strand = df %&gt;% group_by(qid) %&gt;%
        summarize(major.strand=ifelse(sum(sign(qe-qs)*abs(qe-qs))&gt;0, &#39;+&#39;, &#39;-&#39;),
                  maxQ=max(c(qe, qs)))
    merge(df, major.strand) %&gt;% mutate(qs=ifelse(major.strand==&#39;-&#39;, maxQ-qs, qs),
                                       qe=ifelse(major.strand==&#39;-&#39;, maxQ-qe, qe),
                                       qid=factor(qid, levels=levels(rid.o$qid)))
}

mumgp.filt.diag = diagMum(mumgp.filt)

ggplot(mumgp.filt.diag, aes(x=rs, xend=re, y=qs, yend=qe, colour=strand)) +
    geom_segment() + geom_point(alpha=.5) + theme_bw() + 
    facet_grid(qid~., scales=&#39;free&#39;, space=&#39;free&#39;, switch=&#39;y&#39;) +
    theme(strip.text.y=element_text(angle=180, size=5), strip.background=element_blank(),
          legend.position=c(.99,.01), legend.justification=c(1,0),
          axis.text.y=element_blank(), axis.ticks.y=element_blank()) +
    xlab(&#39;reference sequence&#39;) + ylab(&#39;assembly&#39;) + scale_colour_brewer(palette=&#39;Set1&#39;)</code></pre>
<p><img src="../../../../post/2017-09-19-MumplotsWithGgplot2_files/figure-html/diag-1.png" width="960" /></p>
<p>What we were aiming at:</p>
<div class="figure">
<img src="../../../../imgs/mumplot-example.png" />

</div>
<p>Pretty good.</p>
<hr />
<p>To also represent multiple reference regions in separate facets, change the <em>facet_grid</em> commands. Here we have only one reference region but the command would be:</p>
<pre class="r"><code>ggplot(mumgp.filt.diag, aes(x=rs, xend=re, y=qs, yend=qe, colour=strand)) +
    geom_segment() + geom_point(alpha=.5) + theme_bw() + 
    facet_grid(qid~rid, scales=&#39;free&#39;, space=&#39;free&#39;, switch=&#39;y&#39;) +
    theme(strip.text.y=element_text(angle=180, size=5), strip.background=element_blank(),
          legend.position=c(.99,.01), legend.justification=c(1,0),
          axis.text.y=element_blank(), axis.ticks.y=element_blank()) +
    xlab(&#39;reference sequence&#39;) + ylab(&#39;assembly&#39;) + scale_colour_brewer(palette=&#39;Set1&#39;)</code></pre>
<p><img src="../../../../post/2017-09-19-MumplotsWithGgplot2_files/figure-html/rid-1.png" width="960" /></p>
<p>See also <a href="https://github.com/jmonlong/Hippocamplus/issues/2">this GitHub issue</a>.</p>
</div>
<div id="percent-identity-and-coverage" class="section level2">
<h2>Percent identity and coverage</h2>
<p>Another useful MUMmerplot represents the position of each aligned segment and its percent similarity.</p>
<p>This graph could be useful to decide which size/similarity threshold to use when filtering low alignments.</p>
<pre class="r"><code>mumgp %&lt;&gt;% mutate(similarity=1-error/abs(qe-qs))
mumgp.filt %&lt;&gt;% mutate(similarity=1-error/abs(qe-qs))

ggplot(mumgp, aes(x=rs, xend=re, y=similarity, yend=similarity)) + geom_segment() +
    theme_bw() + xlab(&#39;reference sequence&#39;) + ylab(&#39;similarity&#39;) + ggtitle(&#39;All contigs&#39;) +
    ylim(0,1)</code></pre>
<p><img src="../../../../post/2017-09-19-MumplotsWithGgplot2_files/figure-html/simgraph-1.png" width="960" /></p>
<pre class="r"><code>ggplot(mumgp.filt, aes(x=rs, xend=re, y=similarity, yend=similarity)) + geom_segment() +
    theme_bw() + xlab(&#39;reference sequence&#39;) + ylab(&#39;similarity&#39;) +
    ggtitle(&#39;At least 10 Kbp aligned&#39;) + ylim(0,1)</code></pre>
<p><img src="../../../../post/2017-09-19-MumplotsWithGgplot2_files/figure-html/simgraph-2.png" width="960" /></p>
<p>To better highlighted which region in the reference is covered, I annotate each base of the reference with the maximum similarity.</p>
<pre class="r"><code>maxSimilarityDisjoin &lt;- function(df){
  ref.ir = GRanges(&#39;X&#39;, IRanges(df$rs, df$re), similarity=df$similarity)
  ## Efficient clean up of low similarity within high similarity
  step = 1
  while(step&gt;0){
    largealign = ref.ir[head(order(rank(-ref.ir$similarity), rank(-width(ref.ir))),step*1000)]
    ol = findOverlaps(ref.ir, largealign, type=&#39;within&#39;) %&gt;% as.data.frame %&gt;%
        mutate(simW=ref.ir$similarity[queryHits],
               simL=largealign$similarity[subjectHits]) %&gt;% filter(simW&lt;simL)
    if(length(largealign) == length(ref.ir)){
      step = 0
    } else {
      step = step + 1
    }
    ref.ir = ref.ir[-ol$queryHits]
  }
  ## Disjoin and annotate with the max similarity
  ref.dj = disjoin(c(ref.ir, GRanges(&#39;X&#39;, IRanges(min(df$rs), max(df$rs)), similarity=0)))
  ol = findOverlaps(ref.ir, ref.dj) %&gt;% as.data.frame %&gt;%
      mutate(similarity=ref.ir$similarity[queryHits]) %&gt;%
      group_by(subjectHits) %&gt;% summarize(similarity=max(similarity))
  ref.dj$similarity = 0
  ref.dj$similarity[ol$subjectHits] = ol$similarity
  as.data.frame(ref.dj)
}

mumgp.sim = maxSimilarityDisjoin(mumgp)

mumgp.sim %&gt;% select(similarity, start, end) %&gt;% gather(end, pos, 2:3) %&gt;%
    ggplot() + geom_line(aes(x=pos, y=similarity), alpha=.5, color=&#39;red&#39;) + theme_bw() +
    xlab(&#39;reference sequence&#39;) + ylab(&#39;similarity&#39;) + ggtitle(&#39;All contigs&#39;) + ylim(0,1) +
    geom_segment(aes(x=rs, xend=re, y=similarity, yend=similarity), data=mumgp)</code></pre>
<p><img src="../../../../post/2017-09-19-MumplotsWithGgplot2_files/figure-html/simcov-1.png" width="960" /></p>
<pre class="r"><code>ggplot(mumgp.sim) + geom_segment(aes(x=start, xend=end, yend=similarity, y=similarity),
                                 color=&#39;red&#39;, size=2) +
    theme_bw() + xlab(&#39;reference sequence&#39;) + ylab(&#39;similarity&#39;) + ylim(0,1) +
    geom_segment(aes(x=rs, xend=re, y=similarity, yend=similarity), data=mumgp)</code></pre>
<p><img src="../../../../post/2017-09-19-MumplotsWithGgplot2_files/figure-html/simcov-2.png" width="960" /></p>
<p>With this graph we could compare different assemblies or before/after filtering:</p>
<pre class="r"><code>mumgp.filt.sim = maxSimilarityDisjoin(mumgp.filt)

mumgp.filt.m = rbind(mumgp.sim %&gt;% mutate(filter=&#39;before&#39;),
                     mumgp.filt.sim %&gt;% mutate(filter=&#39;after&#39;))

mumgp.filt.m %&gt;% select(similarity, start, end, filter) %&gt;% gather(end, pos, 2:3) %&gt;%
    ggplot(aes(x=pos, y=similarity, colour=filter)) + geom_line(alpha=.8) + theme_bw() +
    xlab(&#39;reference sequence&#39;) + ylab(&#39;similarity&#39;) + ylim(0,1) +
    scale_colour_brewer(palette=&#39;Set1&#39;)</code></pre>
<p><img src="../../../../post/2017-09-19-MumplotsWithGgplot2_files/figure-html/simcovcomp-1.png" width="960" /></p>
<p>Not so pretty but we see that a few region are not covered any more after our filtering. Maybe something like this instead :</p>
<pre class="r"><code>mumgp.filt.m %&gt;% filter(similarity==0) %&gt;%
    ggplot(aes(x=start, xend=end, y=filter, yend=filter)) + geom_segment(size=10) +
    theme_bw() + xlab(&#39;reference sequence&#39;) + ylab(&#39;filter&#39;) +
    scale_colour_brewer(palette=&#39;Set1&#39;) + ggtitle(&#39;Reference regions not covered&#39;)</code></pre>
<p><img src="../../../../post/2017-09-19-MumplotsWithGgplot2_files/figure-html/simcovcomptrack-1.png" width="960" /></p>
</div>

</div>

      </div>
    </div>

    
    

    <label for="sidebar-checkbox" class="sidebar-toggle"></label>

    <script src="../../../../assets/highlight.pack.js"></script>
    <script src="../../../../assets/r.min.js"></script>
    
    <script>
      hljs.configure({languages: ['r']});
      hljs.initHighlightingOnLoad();
    </script>

    <script src="//yihui.name/js/math-code.js"></script>
    <script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
    
  </body>
</html>

